src/mxarray.jl

```diff
@@ -157,13 +158,16 @@ const _mx_is_complex = mxfunc(:mxIsComplex)
 const _mx_is_sparse  = mxfunc(:mxIsSparse)
 const _mx_is_empty   = mxfunc(:mxIsEmpty)
 const _mx_is_struct  = mxfunc(:mxIsStruct)
-const _mx_is_cell    = mxfunc(:mxIsCell) 
+const _mx_is_cell    = mxfunc(:mxIsCell)
 
 
 # getting simple attributes
 
 macro mxget_attr(fun, ret)
-    :( ccall($(fun)::Ptr{Void}, $(ret), (Ptr{Void},), mx.ptr) )
+    quote
+        @assert mx.ptr::Ptr{Void} != C_NULL
```

Comment:
Good point. I added them here so that the  errors can be triggered at the lowest levels, since these functions are called with `size` , `length` etc.

Where should it be an `unsafe_convert`?